gorgias/gorgias-chrome
# Firestore backend

New Firestore backend for template sync and management.
See `FIRESTORE.md` for instructions on how to run it.

### Features:

* Log-in with your regular email and password.
* New backend gets enabled automatically for whitelisted customers (only Gorgias customer is enabled right now).
* Non-whitelisted customers should not notice any changes.
* User accounts use Firebase Auth.
* Template storage uses Firestore.
* Cloud Functions for server-side functionality.
* Supports creating logged-out templates.
* When logging-in, templates created while logged-out are synced and deleted from local storage. Avoids having to implement sync with conflict resolution.

### How it works:

* Created a new `store` that uses a plugin system (Firestore or Old API).
* Plugin gets switched automatically if customer is whitelisted.
* All data is now stored in the background script.
* The content script uses `chrome.runtime.sendMessage` to request data from the background script.
* Moved all old-API interaction to the new old-api-plugin (`store/plugin-api.js`).
* Made as little changes to the Angular app as possible, mostly just to services. The Angular app is now independent from the old API and `TemplateStorage`.


### Still missing:

* Team member management
* Subscription management
* Subscribe new accounts
* Stats


### Bugfixes:
* Fix issues with never sending stats to the api because of not checking the correct setting.
* Fix issues with trying to get sharing for local templates (with no `remote_id`).
* Remove Coupon API, it wasn't used.
* Fix issues with showing `shared with=everyone` when having specific sharing set to n - 2 members.


### Testing:

* See setup instructions in `FIRESTORE.md`.
* Run staging with `yarn staging`.
* Log-in with your regular email and password.
* All your existing templates should be visible.
* All template management functionality should work (CRUD on templates, sharing).

To make sure you're using the new Firestore backend you can check the Settings page. You should see a new `Experimental` section at the bottom of the page when the plugin is enabled.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gorgias/gorgias-chrome/311)
